src/EventStore.Core.Tests/Services/ElectionsService/FakeEpochManager.cs

```diff
@@ -35,26 +35,30 @@ internal class FakeEpochManager : IEpochManager {
 			}
 		}
 
-		public EpochRecord GetEpoch(int epochNumber, bool throwIfNotFound) {
+		public EpochRecord GetEpochAfter(int epochNumber, bool throwIfNotFound) {
 			lock (_epochs) {
 				var epoch = _epochs.FirstOrDefault(e => e.EpochNumber == epochNumber);
+				if (epoch != null) {
+					var index = _epochs.IndexOf(epoch);
```

Comment:
if the epoch is not found, indexOf will return -1 and we should throw an exception here.

Reply:
we're inside a lock on the collection and we just found this epoch by searching the collection, so we can trust that it is still there.